### ISSUE TITLE
Decimal places - data transmitted to influx

### ISSUE THREAD

**Issue by ravsit-activum** [domoticz/domoticz#3960]
https://github.com/domoticz/domoticz/blob/ab401616e317170fc31bf8396c5ce10befa52784/push/BasePush.cpp#L515

If this is not a problem, I suggest increasing the accuracy of the data transmitted to influx.

In: BasePush.cpp
1. for Temperature - line 504:
sprintf (szData, "**% .2f**", tvalue);

2. for humidity - line 515:
sprintf (szData, "**% .2f", atof (rawsendValue.c_str ())**);

**Comment by gizmocuz** [comment]
How about using %g ? This will use a integer for integer and has no rounding issues
Do you have such a accurate temp sensor ? You need to calibrate the sensor as well if you want to have a correct value
Else 1 decimal should be sufficient

**Comment by gizmocuz** [comment]
I changed the code to use %g now, If you compile yourself, could you test this ?
Else it will be available in beta 11761 which should be there for x86/arm systems within the hour.
Currently there is a download issue on the boost website that prevents automatic compilation for travis (ubuntu) builds

**Comment by ravsit-activum** [comment]
Domoticz save data with 2 decimals - so it is better to send it to influx too.
I have sensor with such accuracy.

from devicestatus table:

ID | HardwareID | DeviceID | Unit | Name | Used | Type | SubType | SignalLevel | BatteryLevel | nValue | sValue | LastUpdate
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
30 | 10 | 82030 | 1 | Temp1 | 1 | 80 | 5 | 3 | 100 | 0 | 23.81 | 27.02.2020 20:40
32 | 10 | 82032 | 1 | Temp3 | 1 | 84 | 1 | 7 | 100 | 0 | 21.65;64.89;0;972;0 | 27.02.2020 20:40
65 | 16 | 82065 | 1 | Temp4 | 1 | 82 | 1 | 3 | 100 | 0 | 22.48;30.81;0 | 30.10.2019 13:01
81 | 19 | 130007 | 7 | Airly - Ciśnienie powietrza | 1 | 243 | 26 | 12 | 255 | 0 | 1001.61;0 | 27.02.2020 20:35

for Barometer - line 523:
		sprintf(szData, "%.2f", atof(rawsendValue.c_str()));

I prefere %.2f because of %g can go to string with e+, for example: 5.6e+4

Look at: https://www.quora.com/What-are-the-differences-between-f-e-and-g-format-specifiers-in-C-language

**Comment by gizmocuz** [comment]
%g is also used internally on other places so i think it is best to do this here as well.
of the sensor has one digit, %g will have one digit

**Comment by ravsit-activum** [comment]
I checked. Influx works with IEEE-754 64-bit floating-point numbers.

g% works.

It can only look surprising in the logs.

**Comment by ravsit-activum** [comment]
You missed change for Humidity

line 516:
				sprintf(szData, "%g", std::stof(rawsendValue));

**Comment by gizmocuz** [comment]
Don't think so why ? Humidity is a integer value

**Comment by ravsit-activum** [comment]
It is not true.

I have some humidity sensors that give 2 decimal places.

For example SHT30 - Temperature:21.25Humidity:60.66

**Comment by gizmocuz** [comment]
@ravsit-activum , nice, but domoticz threats all humidity sensors as integer..., it also does not make much sense

**Comment by ravsit-activum** [comment]
I do not agree.
Domoticz stores Humidity data i database with decimals. Look at my table sensor Temp3 - type 84, subtype 1 (sValue=21.65;**64.89**;0;972;0). The second value is Humidity.

For me it is very important to send data to influx with source resolution. 
Domoticz can show humidity as integer and can send data to influx without truncates or roundings.

I have more such humidity sensors in my enviroments. Till now I do dummy temp sensor to keep original 2 decimal source accuracy.

So, It makes sense to me and i think to others.

**Comment by gizmocuz** [comment]
The SHT30 has a Humidity fault tolerance of 2%, anyway it does not make sense.
Throughout the application it is also used as an Integer, and let's keep it that way

**Comment by ravsit-activum** [comment]
Sensors, as a rule, have 2 features: accuracy and resolution.
1. Accuracy/tolerance, e.g +/- 2% - means a margin/difference from real measurement value.
2. Resolution/precision, e.g 0,01% - allows you to measure the difference in measurements with the minimum value difference available - with very good precision.

https://en.wikipedia.org/wiki/File:High_accuracy_Low_precision.svg
vs
https://en.wikipedia.org/wiki/File:High_precision_Low_accuracy.svg

SHT30 has resolution 0,01%
(page 2)
https://www.bmf-systemparts.com/wp-content/uploads/2018/04/Sensirion_Humidity_Sensors_SHT3x_Datasheet_analog.pdf

I have another sensors based on Si7021-A20 - with 12-bit resolution, 0,025%.

In many environments, it's important to see every small change in measurement, so we need these decimal places after dot.

Maybe I can convince You somehow?

Maybe someone else will convince You?

All we need more precision in Humidity. ;-)

**Comment by gizmocuz** [comment]
Well You may create a PR for the change in the push base class if you want,
but internally it is still a integer. Meaning , in the internal graphs you will see whole numbers

That also does not say "I missed change for Humidity" as it's a feature request

I think the BME280 is more accurate then the SHT30, but I don't have such resolution usage for the humidity
That a sensor 'has' more precision, does not say it is needed in a 'home automation system' what this project is all about
Also any 433/868Mhz sensors will not report the humidity with decimals as far as I know

So to recap, please make a PR for the base push class if you would like to have this pushed to InfluxDB/MQTT
